build/dotnet/dotnet-install.sh

```diff
@@ -512,77 +549,117 @@ download() {
     local out_path=${2:-}
 
     local failed=false
-    # Restart the request up to N times if it fails
-    local retries=3
-    # Give up after 120 seconds of retrying
-    local retry_max_time=120
+    if machine_has "curl"; then
+        downloadcurl $remote_path $out_path || failed=true
+    elif machine_has "wget"; then
+        downloadwget $remote_path $out_path || failed=true
+    else
+        failed=true
+    fi
+    if [ "$failed" = true ]; then
+        say_verbose "Download failed: $remote_path"
+        return 1
+    fi
+    return 0
+}
+
+downloadcurl() {
+    eval $invocation
+    local remote_path=$1
+    local out_path=${2:-}
 
+    local failed=false
     if [ -z "$out_path" ]; then
-        curl --fail --retry $retries --retry-max-time $retry_max_time -sSL $remote_path \
-            || wget -qO- $remote_path \
-            || failed=true
+        curl --retry 10 -sSL -f --create-dirs $remote_path || failed=true
```

Comment:
We can keep this in our script for now, but we should eventually create a PR to merge this back into the CLI script.  This will make it easier to keep the scripts in sync.